### ISSUE TITLE
"Solve CAPTCHA to continue" Issue.

### ISSUE THREAD

**Issue by sonugit0607** [vsakkas/sydney.py#166]
I have been using Sydney since a while, and recently in the past week all of a sudden I'm facing "Solve CAPTCHA to continue" issue continuously, whether I clear my cookies and cache. Also tried adding the Copilot cookies in the code itself as mentioned in the **Pypi Documentation**  still didn't work for me. Please help me with this issue, Thanks in advance.

<img width="326" alt="Screenshot 2024-05-09 144346" src="https://github.com/vsakkas/sydney.py/assets/104677341/ce3e8dc8-92d3-41bf-9826-c1ae88f09db3">

**Comment by sonugit0607** [comment]
@vsakkas have a look into it!, it would solve me a big problem if I get any solution. due to this one of my project is on hold🥲.

**Comment by vsakkas** [comment]
Hi @sonugit0607 

Sorry, I have been inactive for quite some time now (been busy moving to another country!), but this seems quite important. I will try to take a look at all the open issues on the repo, including this one, in the next few days :)

**Comment by sonugit0607** [comment]
thanks!

**Comment by usun1997** [comment]
easy question. You need to manually go to copilot website and start one conversation. and then microsoft bing will verify your identity, after verification, you can use the python copilot.

**Comment by sonugit0607** [comment]
tried multiple times, didn't work.

**Comment by sonugit0607** [comment]
it's not happening in my case, I'm using the same accounts cookies to which I have logged in.

**Comment by usun1997** [comment]
so you are able to manually chat with copilot in many rounds on the web, and there is no box comming out to let you click "confirmed" something like that? theoretically， if you can chat on the web then your python  copilot is supposed to work. If you still encounter this issue, I think it is because you account is banned by microsoft for some reason (but I don't believe it). You can test it by creating a new account and see whether python copilot works again.

**Comment by vsakkas** [comment]
Hi @sonugit0607
I haven't been able to reproduce this issue unfortunately. I've released a couple minor updates to the sydney package, so please update to the latest version (although I don't think that will change anything)

Besides that, it is possible that your account or IP address has been blocked. Can you reproduce this issue with a different account or with a different IP or on a different machine?

**Comment by sonugit0607** [comment]
Hi @vsakkas In the same device I have tried with multiple accounts as well, getting same issue. Not sure what's causing this.

**Comment by RexsyBima** [comment]
yup, can confirm this, i got this weird solve captcha to continue even if i already try to change the cookies and everything

**Comment by ammar-n-abbas** [comment]
Similar issue faced by my team too

**Comment by RexsyBima** [comment]
Hey an update here, i manage to get it to working again by changing to other microsoft account, it seems they block my account for interacting with sydney py too many times, i hope the blocking is temporary

**Comment by vsakkas** [comment]
To summarize what I've read so far, here's a few things to try:
- Get new cookies
- Use a different account
- Use a different machine
- Use a different account **and** machine

It is possible that either the IP or the machine have gotten blocked from high usage of the API.

Also, if you're using a proxy, keep in mind that currently proxy support is.. not the best. I will if I can do something about it during this weekend.

**Comment by mizrits** [comment]
I also encountered the same situation. But after updating the cookie value, it works again (without CAPTCHA)!

**Comment by mrkami1** [comment]
I'm getting the same issue and I have tried changing the cookies.

**Comment by vsakkas** [comment]
> To summarize what I've read so far, here's a few things to try:
> 
>     * Get new cookies
> 
>     * Use a different account
> 
>     * Use a different machine
> 
>     * Use a different account **and** machine
> 
> 
> It is possible that either the IP or the machine have gotten blocked from high usage of the API.
> 
> Also, if you're using a proxy, keep in mind that currently proxy support is.. not the best. I will if I can do something about it during this weekend.

Another thing that I didn't mention is that you need to send a message to Copilot on the website itself so that you can verify that you're a human. Usually this shows up on the first message, but after testing, it seems that lately it can also show up on the second message.

**Comment by mrkami1** [comment]
> I'm getting the same issue and I have tried changing the cookies and tried a different machine.

I've now gotten it to work. I was previously using the cookies from Firefox and now tried cookies from Edge and it works fine.

**Comment by Luneye** [comment]
Maybe the retrieved cookies from Firefox aren't working due to the user headers emulation in the constants.py file (the CREATE_HEADERS constant). I believe that if you change the CREATE_HEADERS constant with the same headers you used in the request to retrieve your cookies on Firefox, it may work!

I also confirm that the cookies retrieved using Edge or Chrome (since both are Chromium-based browsers) work with the current implementation of this project :)

**Comment by guhuajun** [comment]
Actually, it's documented. You just need to open bing copilot, click new conversation, and send a simple `hi`, when the verification is ok, you can go to command line. The verification process is short, may difficult to do a screenshot.

```plaintext
[!IMPORTANT] For regions where a cookie is required, it is recommended to manually write messages to Copilot until a box containing a Verifying message appears, which should then switch to a Success! message. Without this step, it is possible that Sydney.py will fail with a CaptchaChallenge error.
```
